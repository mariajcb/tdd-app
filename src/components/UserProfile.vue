<template lang="html">
  <div>
    UserProfile
  </div>
</template>

<script>
export default {
  name: 'UserProfile'
}
</script>
