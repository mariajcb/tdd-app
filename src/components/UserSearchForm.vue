<template lang="html">
  <div>
    UserSearchForm
  </div>
</template>

<script>
export default {
  name: 'UserSearchForm'
}
</script>
