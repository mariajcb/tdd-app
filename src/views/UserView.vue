<template lang="html">
  <div>
    <UserSearchForm @submitted="searchUser"/>
    <UserProfile :user="user" />
  </div>
</template>

<script>

import { mapState } from 'vuex'
import UserSearchForm from '@/components/UserSearchForm'
import UserProfile from '@/components/UserProfile'

export default {
  name: 'UserView',
  components: {
    UserSearchForm,
    UserProfile,
  },
  computed: {
    ...mapState({
      user: 'user',
    })
  },
  methods: {
    searchUser(username) {
      this.$store.dispatch('SEARCH_USER', { username })
    }
  }
}
</script>
