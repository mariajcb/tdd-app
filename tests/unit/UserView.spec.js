import { shallowMount } from '@vue/test-utils'
import UserView from '@/views/UserView'
import UserSearchForm from '@/components/UserSearchForm'
import UserProfile from '@/components/UserProfile'

describe('UserView', () => {
  const build = () => {
    const wrapper = shallowMount(UserView)

    return {
      wrapper,
      const userSearchForm: () => wrapper.find(UserSearchForm),
      const userProfile: () => wrapper.find(UserProfile)
    }
  }
  it('renders the component', () => {
    //arrange
    const { wrapper } = build()
    //assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders main child components', () => {
    //arrange
    const { userSearchForm, userProfile} = build()
    //assert
    expect(userSearchForm.exists()).toBe(true)
    expect(userProfile.exists()).toBe(true)
  })
})
