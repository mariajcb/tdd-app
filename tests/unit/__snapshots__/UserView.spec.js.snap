// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <usersearchform-stub></usersearchform-stub>
  <userprofile-stub user="[object Object]"></userprofile-stub>
</div>
`;
